### PR TITLE
samples: openthread: align sample.yaml configs to CI tests

### PR DESCRIPTION
**Pull request by jaul-nsc**
add missing configurations and revise existing ones for the CI test needs

### CODE REVIEW

File: samples/openthread/cli/snippets/ci/ci.conf [hunk 1]
```diff
@@ -16,3 +16,9 @@ CONFIG_SHELL_DEFAULT_TERMINAL_WIDTH=618
 CONFIG_ASSERT=y
 CONFIG_ASSERT_NO_COND_INFO=y
 CONFIG_RESET_ON_FATAL_ERROR=n
+
+# Enable Zperf
+CONFIG_NET_SHELL=y
+CONFIG_NET_ZPERF=y
+CONFIG_NET_CONTEXT_RCVTIMEO=y
+CONFIG_NET_SOCKETS=y
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/snippets/ci/ci.conf:24]:
```suggestion
CONFIG_NET_SOCKETS=y
CONFIG_NET_SOCKETS_POLL_MAX=9
```
It is now needed for Zperf  APIs

File: samples/openthread/cli/sample.yaml [hunk 2]
```diff
@@ -1,77 +1,113 @@
...
+    tags: ci_build
+    platform_allow: >
+      nrf52840dk_nrf52840
+    extra_args: >
+      SNIPPET="ci;logging;multiprotocol"
+      CONFIG_OPENTHREAD_BORDER_ROUTER=y
+      CONFIG_OPENTHREAD_BORDER_AGENT=y
+      CONFIG_OPENTHREAD_BACKBONE_ROUTER=y
+      CONFIG_OPENTHREAD_BACKBONE_ROUTER_DUA_NDPROXYING=y
+      CONFIG_OPENTHREAD_BACKBONE_ROUTER_MULTICAST_ROUTING=y
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/sample.yaml:84]: Should we put this in new snippet?

File: samples/openthread/cli/sample.yaml [hunk 3]
```diff
@@ -1,77 +1,113 @@
...
   name: OpenThread CLI sample
   description: Test OpenThread Command Line Interface.
+
 tests:
-  sample.openthread.cli:
+  sample.openthread.cli.multiprotocol:
     build_only: true
-    platform_allow: >
-      nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp_ns nrf52840dk_nrf52840
-      nrf52840dongle_nrf52840 nrf21540dk_nrf52840
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/sample.yaml:9]: I think we should keep `nrf52840dongle_nrf52840` targets?

File: samples/openthread/cli/sample.yaml [hunk 4]
```diff
@@ -1,77 +1,113 @@
...
-    platform_allow: >
-      nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp_ns nrf52840dk_nrf52840 nrf21540dk_nrf52840
     tags: ci_build
+    platform_allow: >
+      nrf52840dk_nrf52840
     extra_args: >
-      SNIPPET="usb;ci;multiprotocol;tcp"
+      SNIPPET="ci;logging;multiprotocol;usb"
     integration_platforms:
-      - nrf5340dk_nrf5340_cpuapp
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/sample.yaml:40]: usb build targets should be restored as well

File: samples/openthread/cli/sample.yaml (hunk 3 above)
Comment (jaul-nsc) [reply samples/openthread/cli/sample.yaml:9]: nrf52840dongle_nrf52840 back in platform_allow

File: samples/openthread/cli/sample.yaml (hunk 4 above)
Comment (jaul-nsc) [reply samples/openthread/cli/sample.yaml:40]: restored

File: samples/openthread/cli/sample.yaml [hunk 5]
```diff
@@ -1,77 +1,113 @@
...
+    extra_args: >
+      SNIPPET="ci;logging"
+    integration_platforms:
       - nrf52840dk_nrf52840
-      - nrf52840dongle_nrf52840
       - nrf21540dk_nrf52840
+      - nrf5340dk_nrf5340_cpuapp
+      - nrf5340dk_nrf5340_cpuapp_ns
+
   sample.openthread.cli.thread_1_1:
```
Comment (canisLupus1313) [inline samples/openthread/cli/sample.yaml:38]: Entire Thread 1.1 examples and building samples could be dropped.

File: samples/openthread/cli/sample.yaml [hunk 6]
```diff
@@ -1,77 +1,113 @@
...
+
   sample.openthread.cli.usb:
     build_only: true
-    platform_allow: >
-      nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp_ns nrf52840dk_nrf52840 nrf21540dk_nrf52840
     tags: ci_build
+    platform_allow: >
+      nrf52840dk_nrf52840
     extra_args: >
-      SNIPPET="usb;ci;multiprotocol;tcp"
```
Comment (canisLupus1313) [inline samples/openthread/cli/sample.yaml:38]: Why are You dropping TCP in so many places?

File: samples/openthread/cli/sample.yaml (hunk 2 above)
Comment (canisLupus1313) [reply samples/openthread/cli/sample.yaml:84]: This part should be dropped from here. It suggests that we are supporting Border Routing functionalities in our samples. While it's just test env workaround.
```suggestion
      CONFIG_OPENTHREAD_BORDER_ROUTER=y
      CONFIG_OPENTHREAD_BORDER_AGENT=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER_DUA_NDPROXYING=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER_MULTICAST_ROUTING=y
```

Comment (canisLupus1313) [reply samples/openthread/cli/sample.yaml:84]:
```suggestion
      CONFIG_OPENTHREAD_BORDER_ROUTER=y
      CONFIG_OPENTHREAD_BORDER_AGENT=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER_DUA_NDPROXYING=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER_MULTICAST_ROUTING=y
```

Comment (canisLupus1313) [inline samples/openthread/cli/sample.yaml:84]:
```suggestion
      SNIPPET="ci;logging;multiprotocol"
      CONFIG_OPENTHREAD_BORDER_ROUTER=y
      CONFIG_OPENTHREAD_BORDER_AGENT=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER_DUA_NDPROXYING=y
      CONFIG_OPENTHREAD_BACKBONE_ROUTER_MULTICAST_ROUTING=y
```

File: samples/openthread/cli/sample.yaml [hunk 7]
```diff
@@ -1,77 +1,113 @@
...
+    platform_allow: >
+      nrf52840dk_nrf52840
+    extra_args: >
+      SNIPPET="ci;logging;multiprotocol"
+      CONFIG_OPENTHREAD_BORDER_ROUTER=y
+      CONFIG_OPENTHREAD_BORDER_AGENT=y
+      CONFIG_OPENTHREAD_BACKBONE_ROUTER=y
+      CONFIG_OPENTHREAD_BACKBONE_ROUTER_DUA_NDPROXYING=y
+      CONFIG_OPENTHREAD_BACKBONE_ROUTER_MULTICAST_ROUTING=y
+
```
Comment (canisLupus1313) [inline samples/openthread/cli/sample.yaml:85]: Shoulld be removed Look comment above.
```suggestion

```

**Review by canisLupus1313 — CHANGES_REQUESTED** [review]
```suggestion

```

File: samples/openthread/cli/snippets/ci/ci.conf (hunk 1 above)
Comment (jaul-nsc) [reply samples/openthread/cli/snippets/ci/ci.conf:24]: done

File: samples/openthread/coap_client/sample.yaml [hunk 8]
```diff
@@ -1,26 +1,46 @@
...
+      nrf5340dk_nrf5340_cpuapp
+      nrf5340dk_nrf5340_cpuapp_ns
+    extra_args: >
+      SNIPPET="ci;logging"
     integration_platforms:
+      - nrf52840dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
       - nrf5340dk_nrf5340_cpuapp_ns
-      - nrf52840dk_nrf52840
-      - nrf21540dk_nrf52840
```
Comment (maciejbaczmanski) [inline samples/openthread/coap_client/sample.yaml:12]: should `nrf21540dk_nrf52840`  target be restored for coap_client samples as well?

File: samples/openthread/coprocessor/sample.yaml [hunk 9]
```diff
@@ -1,33 +1,46 @@
...
     build_only: true
-    extra_args: SNIPPET="ci;usb"
+    tags: ci_build
+    platform_allow: >
+      nrf52833dk_nrf52833
+      nrf52840dk_nrf52840
+      nrf21540dk_nrf52840
+    extra_args: >
+      SNIPPET="ci;usb"
+      CONFIG_OPENTHREAD_DIAG=y
```
Comment (maciejbaczmanski) [inline samples/openthread/coprocessor/sample.yaml:31]: Why is this config needed here? Could it be added to `ci` snippet?

**Review by canisLupus1313 — APPROVED** [review]

File: samples/openthread/coprocessor/sample.yaml (hunk 9 above)
Comment (jaul-nsc) [reply samples/openthread/coprocessor/sample.yaml:31]: Thanks for catching that, I left it by mistake.

**Review by maciejbaczmanski — APPROVED** [review]

**Pull request merged**